### PR TITLE
refine detection of PPM URLs

### PR DESCRIPTION
**Pull request by kevinushey**
Closes https://github.com/rstudio/renv/issues/445.

### CODE REVIEW

File: R/ppm.R [hunk 1]
```diff
@@ -33,35 +33,34 @@ renv_ppm_transform_impl <- function(url) {
...
-  # any local knowledge of available CRAN repositories)
+  # check if this is an 'ignored' URL; that is, a repository which we
+  # know is not a PPM URL
   mirrors <- catch(getCRANmirrors(local.only = TRUE))
-  urls <- c(urls, mirrors$URL)
-
-  # also ignore some RStudio URLs
-  rstudio <- c(
+  ignored <- c(
+    getOption("renv.ppm.ignoredUrls", default = character()),
```
Comment (hadley) [inline R/ppm.R:55]: Need to document `renv.ppm.ignoredUrls` somewhere?

**Comment by aronatkins** [comment]
CC @toph-allen - we may want this change.

File: tests/testthat/test-ppm.R [hunk 2]
```diff
@@ -79,3 +79,34 @@ test_that("renv correctly detects RHEL as CentOS for RSPM", {
...
+    renv_condition_signal("bail")
+  ))
+
+  # should exit early
+  status <- tryCatch(
+    renv_ppm_transform_impl("http://foo/bar"),
+    bail = identity
+  )
+
+  expect_identical(status, "http://foo/bar")
```
Comment (hadley) [inline tests/testthat/test-ppm.R:102]:
```suggestion
  status <- expect_no_condition(renv_ppm_transform_impl("http://foo/bar"), class = "bail")
  expect_equal(status, "http://foo/bar")
```
?

File: tests/testthat/test-ppm.R [hunk 3]
```diff
@@ -79,3 +79,34 @@ test_that("renv correctly detects RHEL as CentOS for RSPM", {
...
+
+  # should emit a condition now
+  status <- tryCatch(
+    renv_ppm_transform_impl("http://foo/bar/baz"),
+    bail = identity
+  )
+
+  expect_s3_class(status, "bail")
```
Comment (hadley) [inline tests/testthat/test-ppm.R:110]:
```suggestion
  expect_condition(renv_ppm_transform_impl("http://foo/bar/baz"), class = "bail"))
```

File: R/url.R [hunk 4]
```diff
@@ -0,0 +1,33 @@
+
+renv_url_parse <- function(url) {
```
Comment (hadley) [inline R/url.R:2]: It would make me feel happier if this had a few tests.

**Review by hadley — APPROVED** [review]

File: R/ppm.R (hunk 1 above)
Comment (kevinushey) [reply R/ppm.R:55]: Or should I just add it to config.yml? Or should it be a project-specific setting?

Comment (hadley) [reply R/ppm.R:55]: Seems like project specific might be most useful?

File: R/url.R (hunk 4 above)
Comment (kevinushey) [reply R/url.R:2]: Added a couple -- should I try to chase down any edge cases?

File: R/ppm.R (hunk 1 above)
Comment (kevinushey) [reply R/ppm.R:55]: I'll do this in a separate PR (I think the other PPM stuff I added should also be project specific)

**Pull request merged**